Makefile

```diff
@@ -1,28 +1,61 @@
-ifndef GOPATH
-$(error No GOPATH set)
-endif
+SHORT_NAME := helm
+DEIS_REGISTRY ?= ${DEV_REGISTRY}
+IMAGE_PREFIX ?= helm
+
+REPO_PATH := github.com/helm/${SHORT_NAME}
 
+# The following variables describe the containerized development environment
+# and other build options
 BIN_DIR := bin
 DIST_DIR := _dist
 GO_PACKAGES := action chart config dependency log manifest release plugins/sec plugins/example codec
 MAIN_GO := helm.go
 HELM_BIN := $(BIN_DIR)/helm
 PATH_WITH_HELM = PATH="$(shell pwd)/$(BIN_DIR):$(PATH)"
-
 VERSION := $(shell git describe --tags --abbrev=0 2>/dev/null)+$(shell git rev-parse --short HEAD)
+DEV_ENV_IMAGE := quay.io/deis/go-dev:0.9.0
+DEV_ENV_WORK_DIR := /go/src/${REPO_PATH}
+DEV_ENV_CMD := docker run --rm -v ${CURDIR}:${DEV_ENV_WORK_DIR} -w ${DEV_ENV_WORK_DIR} ${DEV_ENV_IMAGE}
+DEV_ENV_CMD_INT := docker run -it --rm -v ${CURDIR}:${DEV_ENV_WORK_DIR} -w ${DEV_ENV_WORK_DIR} ${DEV_ENV_IMAGE}
+LDFLAGS := "-s -X main.version=${VERSION}"
 
-export GO15VENDOREXPERIMENT=1
 
-ifndef VERSION
-  VERSION := git-$(shell git rev-parse --short HEAD)
-endif
+
+
+# Allow developers to step into the containerized development environment
+
+# Containerized dependency resolution
+bootstrap: check-docker
+	${DEV_ENV_CMD} glide install
+
+# Containerized build of the binary
+dist: check-docker
+	mkdir -p ${BIN_DIR}
+	${DEV_ENV_CMD} make binary-build
```

Comment:
`binary-build` is already run inside a container, so this command will fail. it should run the target, not in a container